web-js/4.x/Search_Widget_3D/README.md

```diff
@@ -0,0 +1,87 @@
+```javascript
+var searchWidget = new Search({
+    visible: true,
+
+    // one of the new awesome features of 4.0 is the separation of the styling and the business logic, such that
+    // all widgets now have their own ViewModel, similar to what is shown below
+    // https://developers.arcgis.com/javascript/latest/api-reference/esri-widgets-Search-SearchViewModel.html
+    viewModel: new SearchVM({
+      view: view, // reference to the SceneView
+      autoNavigate: true, // automatically navigate to the selected result once selected
+      maxResults: 5, // the maximum number of search results to return, the default value is 6     
+      maxSuggestions: 3, // maximum number of suggestions returned by the widget
+      popupEnabled: true, // display the Popup when the feature is clicked
+      popupOpenOnSelect: false, // will not show the Popup when a result is selected from the Search results
+
+      // define which services will be used for the search
+      // https://developers.arcgis.com/javascript/latest/api-reference/esri-widgets-Search-SearchViewModel.html#sources
+      sources: [{
+        locator: new Locator("//geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer"),
+        singleLineFieldName: "SingleLine",
+        outFields: ["Addr_type"],
+        placeholder: "JavaScript 4.0",
+        resultSymbol: new PictureMarkerSymbol({
+           url: "http://noashx.github.io/img/support.png",
+           size: 50,
+           width: 50,
+           height: 50,
+           xoffset: 0,
+           yoffset: 0
+        }) // closes resultSymbol
+      }] // closes sources
+    }) // closes viewModel
+ }); // closes searchWidget
+```
+The actual widget constructor was used for naming the widget and setting the visibility to true (default is true anyway). The SearchViewModel handles all the configurations that were set to customize the user experience and interface the way I needed (see the sample for comments). What’s interesting about this ViewModel system is that it’s now way easier to share business logic amongst widgets and code bases, because it’s separated from the widget constructor. Here is some more information about the SearchViewModel here: https://developers.arcgis.com/javascript/latest/api-reference/esri-widgets-Search-SearchViewModel.html
+
+The second interesting aspect of 4.0 I will discuss here is how we add the Search widget to the app. Let’s look at the code snippet from 4.0 and from 3.x. Currently, we can add the widget directly to the view’s user interface without creating a separate div element. This is an efficient (and dare I say, elegant) workflow.
+
+```javascript
+//4.x
```

Comment:
@bsnider inline code comments are a nifty way to drill down to the exact location of your own suggestion. since this is already merged, feel free to tweak in your own PR!